readyset-errors/src/lib.rs

```diff
@@ -622,6 +622,9 @@ pub enum ReadySetError {
     #[error("Change in DDL requires partial resnapshot")]
```

Comment:
Can we add doc comments to `ResnapshotNeeded` and `FullResnapshotNeeded`? Specifically, I think we should be clear that `FullResnapshotNeeded` will definitively trigger a full resnapshot, whereas `ResnapshotNeeded` _may_ trigger a full resnapshot depending on whether every table has a replication offset

Reply:
Doc comments are just regular code comments with three `/`s. Something like this (though please tweak my wording):

```suggestion
    /// A DDL change in the upstream database requires resnapshotting some tables. This *may* trigger a full resnapshot depending on whether every table has a replication offset
    #[error("Change in DDL requires partial resnapshot")]
```

Reply:
I think this change and @ethan-readyset 's https://github.com/readysettech/readyset/pull/552/files#diff-6b3555f477ada065cd5bde8d783723350fcdd4302a0b11bdec45b4e034a2326dR622 will conflict. I will submit it today, but I can rebase my branch once his changes land.

Reply:
You should go ahead and merge this first once it's approved -- my plan was to rebase after you merge this and make use of the new `FullResnapshotNeeded` error variant

Reply:
Done